query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java

```diff
@@ -58,32 +81,56 @@ public PostgresClient getPostgresClient(String clusterName) {
 
   public static class PostgresClient {
 
-    private final Connection connection;
+    private final DataSource dataSource;
 
     private PostgresClient(RequestHandlerClientConfig clientConfig) throws SQLException {
       String user = clientConfig.getUser().orElseThrow(IllegalArgumentException::new);
       String password = clientConfig.getPassword().orElseThrow(IllegalArgumentException::new);
-      String url =
-          String.format(
-              "%s?user=%s&password=%s", clientConfig.getConnectionString(), user, password);
-      LOG.debug(
-          "Trying to create a Postgres client connected to postgres server using url: {}", url);
-      this.connection = DriverManager.getConnection(url);
+      String url = clientConfig.getConnectionString();
+      this.dataSource = createPooledDataSource(url, user, password);
     }
 
-    private PostgresClient(Connection connection) {
-      this.connection = connection;
+    private DataSource createPooledDataSource(String url, String user, String password) {
+      LOG.debug(
+          "Trying to create a Postgres client connected to postgres server using url: {}, user: {}, password: {}",
+          url,
+          user,
+          password);
+      ConnectionFactory connectionFactory = new DriverManagerConnectionFactory(url, user, password);
+      PoolableConnectionFactory poolableConnectionFactory =
+          new PoolableConnectionFactory(connectionFactory, null);
+      GenericObjectPool<PoolableConnection> connectionPool =
+          new GenericObjectPool<>(poolableConnectionFactory);
+      connectionPool.setMaxTotal(50);
```

Comment:
Once we are fine with current approach, I will move these values to config

Reply:
yes, we will need that. 

Prior, we were having one connection/handler. With this, we will have a pool of connections/handler. This will require an appropriate config for PgPool + max_connection setting on the server.

What are the defaults values?